src/lib/assignment.ts

```diff
@@ -131,16 +139,36 @@ async function loadYaml(yamlDir: string): Promise<Yaml[]> {
   return validityState(res)
 }
 
-async function reducePublish(courseId: string, srcDir: string, yamlDir: string, changelog: string): Promise<void> {
+async function reducePublish(courseId: string, srcDir: string, yamlDir: string, changelog: string, courseModules: undefined | Module[]): Promise<void> {
   const ymlCfg = await loadYaml(yamlDir)
   for(const item of ymlCfg) {
     console.log(`publishing ${JSON.stringify(item)}`)
     const tmpDstDir = fs.mkdtempSync('/tmp/publish_codio_reduce')
     const paths = item.paths || []
     paths.push(`!${yamlDir}`) // exclude yaml directory from export
     paths.push(`!${yamlDir}/**`) // exclude yaml directory from export
+    
+    let assignmentId
+    if (item.assignmentName && courseModules !== undefined) {
```

Comment:
can you clarify why `courseModules` array is needed here and what for?

Reply:
need to find assignment by name in course. because yaml files are parsed here and assignment name is known only here we use courseModules.